### DISCUSSION TITLE
how to set {language} dynamically in reply.view using fastify

### DISCUSSION THREAD

**Discussion by vijayShankarfastify** [fastify/fastify#3489] [General]
Hi ,  we have express application where we are getting browser language and setting in response renderer as follow

```
    app.get('/base/oj-cgbu-omc-landing-page/index.html', setLanguage);
   function setLanguage(req, res){
    let lang="en-US"
    if(Array.isArray(req.acceptsLanguages())){
        lang=req.acceptsLanguages()[0];
    }
    res.render('index.html', { language: lang });
}
```

 can some body suggest how to replace above with fastify.

**Comment by vijayShankarfastify** [comment]
i am trying in my example as below but it is not raplacing {language} in html file

app.get('/base/oj-cgbu-omc-landing-page/index.html', (req, reply) => {
  reply.view('/web/public/base/oj-cgbu-omc-landing-page/index.html', { language: 'en-US' })
})

**Reply by vijayShankarfastify** [reply]
any suggestions

**Reply by Eomm** [reply]
You need to use a template engine to change variables in your HTML page.

Which engine are you using?
If you try the doc example here
https://github.com/fastify/point-of-view/#point-of-view

you should be able to arrange a proof of concepts

**Reply by vijayShankarfastify** [reply]
I am using ejs engine
app.register(require('point-of-view'), {
    engine: {
      ejs: ejs
    }
  })
app.get('/', function (req, reply) {
reply.view('hello.html', {language: 'en-US'} )
});

i tried like above for replacing {language} variable inside 'hello.html', but it is not replacing.
can u give example, how to set.

**Reply by Eomm** [reply]
You can check the tests

https://github.com/fastify/point-of-view/blob/master/test/test-ejs.js

**Reply by vijayShankarfastify** [reply]
https://github.com/fastify/point-of-view/blob/master/test/test-ejs.js#L283

Not getting which text we are replacing with what text in index-bare.html

could you please eloboarte that test case

**Reply by vijayShankarfastify** [reply]
server.js

const data = { language: 'en-US' }
var ejs = require('ejs');

app.register(require('point-of-view'), {
    engine: {
      ejs: ejs
    }
  })

app.addHook('preHandler', function (request, reply, done) {
    reply.locals = localsData
    done()
  })
app.get('/', function (req, reply) {
reply.view('hello.html', data)
    });

hello.hml


html lang='{{language}}'

    <head>
        <meta charset="UTF-8">
        <title>Hello!</title>
    </head>

    <body>
        <h1>Hello World! text</h1>
        <p>This is a simple paragraph.</p>
    </body>

</html>

===============
i wrote code like above to replace {language} variable in hello.html with 'en-US' . but that is not working.
could you please add your comments

**Reply by vijayShankarfastify** [reply]
@Eomm could you please add your comments

**Reply by vijayShankarfastify** [reply]
i tried with 
server.js
reply.view('hello.html', { text: 'en-US123' })
hello.html
<html lang=<%= text %>>
  <head></head>
  <body>
    <p></p>

    <br/>
  </body>
</html>

It is working, but is<%= text %> mandatory  to have <%= %> syambols and text keyword. 
I tried with {{language}} or <%= language %> but they are not working

**Reply by Eomm** [reply]
I would suggest reading the ejs documentation https://ejs.co/ since the `{{language}}` is not supported for sure.

**Reply by vijayShankarfastify** [reply]
Got the clue   and resolved  using  marko  npm module  . for any custom tag issues which marko is failing to parse  maintain marko.json where we will deaclare all our custom tags.                                                                                                                                                                                                                                                                                                                       function setLanguage (req, reply) {
    let lang = 'en-US'; // default language
    var acceptLanguage = req.headers['accept-language'];
    if (acceptLanguage) {
        lang = acceptLanguage.split(',')[0];
        console.log('language Inside loop is:::'+ lang);
    }
    console.log('language is:::'+ lang);
    reply.view('hello.marko.html', { language: lang });
  }

app.get('/', setLanguage);    

Thank you @Eomm for your help on this.
